### ISSUE TITLE
`build` fails in SAM 1.20.0 with nested stack

### ISSUE THREAD

**Issue by paulie4** [aws/aws-sam-cli#2692]
### Description:
SAM 1.20.0 is not able to `build --use-container` if a local nested application/stack has a Python `AWS::Serverless::Function`.

### Steps to reproduce:
1. `sam init` the Python "Hello World Example" AWS quick start application template in the default `sam-app` directory
2. `mkdir sam-app-top-level`
3. create a `sam-app-top-level/template.yaml` with this content (assuming `sam-app-top-level` is in the same directory as `sam-app`):
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Resources:
  MainStack:
    Type: AWS::Serverless::Application
    Properties:
      Location: ../sam-app/template.yaml
```
4. `cd sam-app-top-level`
5. `sam build --use-container --debug`

### Observed result:
```
2021-03-06 00:37:31,498 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics
2021-03-06 00:37:31,764 | 'build' command is called
2021-03-06 00:37:31,780 | Starting Build inside a container
2021-03-06 00:37:31,785 | No Parameters detected in the template
2021-03-06 00:37:31,811 | 1 stacks found in the template
2021-03-06 00:37:31,842 | No Parameters detected in the template
2021-03-06 00:37:31,873 | 2 stacks found in the template
2021-03-06 00:37:31,873 | No Parameters detected in the template
2021-03-06 00:37:31,904 | 1 resources found in the stack
2021-03-06 00:37:31,904 | No Parameters detected in the template
2021-03-06 00:37:31,937 | 2 resources found in the stack MainStack
2021-03-06 00:37:31,937 | No Parameters detected in the template
2021-03-06 00:37:31,967 | No Parameters detected in the template
2021-03-06 00:37:31,983 | No Parameters detected in the template
2021-03-06 00:37:32,014 | Found Serverless function with name='HelloWorldFunction' and CodeUri='hello_world/'
2021-03-06 00:37:32,014 | No Parameters detected in the template
2021-03-06 00:37:32,045 | No Parameters detected in the template
2021-03-06 00:37:32,061 | Instantiating build definitions
2021-03-06 00:37:32,076 | Same function build definition found, adding function (Previous: BuildDefinition(python3.8, hello_world/, Zip, , 2f61dfe4-60a6-4973-a3bd-7c8300e4a234, {}, {}, []), Current: BuildDefinition(python3.8, hello_world/, Zip, , 45e010da-992d-4d41-b6d8-e845a09521c0, {}, {}, []), Function: Function(name='HelloWorldFunction', functionname='HelloWorldFunction', runtime='python3.8', memory=None, timeout=3, handler='app.lambda_handler', imageuri=None, packagetype='Zip', imageconfig=None, codeuri='hello_world/', environment=None, rolearn=None, layers=[], events={'HelloWorld': {'Type': 'Api', 'Properties': {'Path': '/hello', 'Method': 'get', 'RestApiId': 'ServerlessRestApi'}}}, metadata=None, inlinecode=None, codesign_config_arn=None, stack_path='MainStack'))
2021-03-06 00:37:32,076 | Building codeuri: hello_world/ runtime: python3.8 metadata: {} functions: ['MainStack/HelloWorldFunction']
2021-03-06 00:37:32,076 | Building to following folder C:\temp\sam-app-top-level\.aws-sam\build\MainStack\HelloWorldFunction

Fetching amazon/aws-sam-cli-build-image-python3.8 Docker container image......
2021-03-06 00:37:32,867 | Mounting C:\temp\sam-app-top-level\hello_world as /tmp/samcli/source:ro,delegated inside runtime container
Using the request object from command line argument
Loading workflow module 'aws_lambda_builders.workflows'
Registering workflow 'PythonPipBuilder' with capability 'Capability(language='python', dependency_manager='pip', application_framework=None)'
Registering workflow 'NodejsNpmBuilder' with capability 'Capability(language='nodejs', dependency_manager='npm', application_framework=None)'
Registering workflow 'RubyBundlerBuilder' with capability 'Capability(language='ruby', dependency_manager='bundler', application_framework=None)'
Registering workflow 'GoDepBuilder' with capability 'Capability(language='go', dependency_manager='dep', application_framework=None)'
Registering workflow 'GoModulesBuilder' with capability 'Capability(language='go', dependency_manager='modules', application_framework=None)'
Registering workflow 'JavaGradleWorkflow' with capability 'Capability(language='java', dependency_manager='gradle', application_framework=None)'
Registering workflow 'JavaMavenWorkflow' with capability 'Capability(language='java', dependency_manager='maven', application_framework=None)'
Registering workflow 'DotnetCliPackageBuilder' with capability 'Capability(language='dotnet', dependency_manager='cli-package', application_framework=None)'
Registering workflow 'CustomMakeBuilder' with capability 'Capability(language='provided', dependency_manager=None, application_framework=None)'
Found workflow 'PythonPipBuilder' to support capabilities 'Capability(language='python', dependency_manager='pip', application_framework=None)'
Running workflow 'PythonPipBuilder'
Running PythonPipBuilder:ResolveDependencies
PythonPipBuilder:ResolveDependencies failed
Traceback (most recent call last):
  File "/usr/local/opt/sam-cli/lib/python3.8/site-packages/aws_lambda_builders/workflows/python_pip/actions.py", line 39, in execute
    package_builder.build_dependencies(
  File "/usr/local/opt/sam-cli/lib/python3.8/site-packages/aws_lambda_builders/workflows/python_pip/packager.py", line 135, in build_dependencies
    raise RequirementsFileNotFoundError(requirements_path)
aws_lambda_builders.workflows.python_pip.packager.RequirementsFileNotFoundError: Requirements file not found: /tmp/samcli/source/requirements.txt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/opt/sam-cli/lib/python3.8/site-packages/aws_lambda_builders/workflow.py", line 269, in run
    action.execute()
  File "/usr/local/opt/sam-cli/lib/python3.8/site-packages/aws_lambda_builders/workflows/python_pip/actions.py", line 45, in execute
    raise ActionFailedError(str(ex))
aws_lambda_builders.actions.ActionFailedError: Requirements file not found: /tmp/samcli/source/requirements.txt
Builder workflow failed
Traceback (most recent call last):
  File "/usr/local/opt/sam-cli/lib/python3.8/site-packages/aws_lambda_builders/workflows/python_pip/actions.py", line 39, in execute
    package_builder.build_dependencies(
  File "/usr/local/opt/sam-cli/lib/python3.8/site-packages/aws_lambda_builders/workflows/python_pip/packager.py", line 135, in build_dependencies
    raise RequirementsFileNotFoundError(requirements_path)
aws_lambda_builders.workflows.python_pip.packager.RequirementsFileNotFoundError: Requirements file not found: /tmp/samcli/source/requirements.txt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/opt/sam-cli/lib/python3.8/site-packages/aws_lambda_builders/workflow.py", line 269, in run
    action.execute()
  File "/usr/local/opt/sam-cli/lib/python3.8/site-packages/aws_lambda_builders/workflows/python_pip/actions.py", line 45, in execute
    raise ActionFailedError(str(ex))
aws_lambda_builders.actions.ActionFailedError: Requirements file not found: /tmp/samcli/source/requirements.txt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/opt/sam-cli/lib/python3.8/site-packages/aws_lambda_builders/__main__.py", line 117, in main
    builder.build(
  File "/usr/local/opt/sam-cli/lib/python3.8/site-packages/aws_lambda_builders/builder.py", line 125, in build
    return workflow.run()
  File "/usr/local/opt/sam-cli/lib/python3.8/site-packages/aws_lambda_builders/workflow.py", line 76, in wrapper
    func(self, *args, **kwargs)
  File "/usr/local/opt/sam-cli/lib/python3.8/site-packages/aws_lambda_builders/workflow.py", line 276, in run
    raise WorkflowFailedError(workflow_name=self.NAME, action_name=action.NAME, reason=str(ex))
aws_lambda_builders.exceptions.WorkflowFailedError: PythonPipBuilder:ResolveDependencies - Requirements file not found: /tmp/samcli/source/requirements.txt
2021-03-06 00:37:33,870 | Build inside container returned response {"jsonrpc": "2.0", "id": 1, "error": {"code": 400, "message": "PythonPipBuilder:ResolveDependencies - Requirements file not found: /tmp/samcli/source/requirements.txt"}}

Build Failed
2021-03-06 00:37:34,300 | Sending Telemetry: {'metrics': [{'commandRun': {'requestId': 'ef69a9e5-8369-4b54-832d-52e37edf7d2c', 'installationId': '85c62234-be33-4ef3-9b62-c0ee1b2a1903', 'sessionId': '8c3616b5-c18d-47c6-85ec-601ca9e7f29f', 'executionEnvironment': 'CLI', 'pyversion': '3.8.8', 'samcliVersion': '1.20.0', 'awsProfileProvided': False, 'debugFlagProvided': True, 'region': '', 'commandName': 'sam build', 'duration': 2790, 'exitReason': 'BuildInsideContainerError', 'exitCode': 1}}]}
2021-03-06 00:37:34,657 | Telemetry response: 200
Error: PythonPipBuilder:ResolveDependencies - Requirements file not found: /tmp/samcli/source/requirements.txt
```

### Expected result:
Build should work.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Windows 10
2. `sam --version`: SAM CLI, version 1.20.0

**Comment by Jeff17Robbins** [comment]
If it helps add some context to this easily reproduced bug, before running `sam build --use-container`, my folder tree looks like this

```
SAM-APP-TOP-LEVEL
├───events
├───hello_world
├───shared
│   └───app-with-layer
│       ├───events
│       ├───hello_world
│       ├───my_test_layer
│       └───tests
│           ├───integration
│           └───unit
└───tests
    ├───integration
    └───unit
```

Running `sam build --use-container` produces this error

```
Mounting C:\Users\jeffr\Documents\projects\svn\Projects\Jeff-AWS\sam\sam-app-top-level\my_test_layer as /tmp/samcli/source:ro,delegated inside runtime container

Build Failed
Error: PythonPipBuilder:ResolveDependencies - Requirements file not found: /tmp/samcli/source/requirements.txt
```

The unexpected side-effect of the failed build is that the nested application's layer subfolder (`my-test-layer`) now shows up at the top level folder, empty.  I'm assuming that this folder was never meant to be created there?

```
SAM-APP-TOP-LEVEL
├───.aws-sam
│   └───build
│       └───MainStack
│           └───HelloWorldFunction
│               ├───certifi
│               ├───certifi-2020.12.5.dist-info
│               ├───chardet
│               │   ├───cli
│               │   └───metadata
│               ├───chardet-4.0.0.dist-info
│               ├───idna
│               ├───idna-2.10.dist-info
│               ├───requests
│               ├───requests-2.25.1.dist-info
│               ├───urllib3
│               │   ├───contrib
│               │   │   └───_securetransport
│               │   ├───packages
│               │   │   ├───backports
│               │   │   └───ssl_match_hostname
│               │   └───util
│               └───urllib3-1.26.3.dist-info
├───events
├───hello_world
├───my_test_layer <<--- empty folder created by `sam build --use-container`
├───shared
│   └───app-with-layer
│       ├───events
│       ├───hello_world
│       ├───my_test_layer
│       └───tests
│           ├───integration
│           └───unit
└───tests
    ├───integration
    └───unit
```

**Comment by aahung** [comment]
Hi @paulie4, thanks for reporting, there is a bug in v1.20.0 causing nested build to fail if the child template. is located in a different folder from the root template, it is being fixed by #2690, current workaround: 
- place templates in the same folder (which is not very cool)
- use path relative to root template for `CodeUri` in child templates (which will be broken when #2690 is released)
- use absolute `CodeUri` in child templates

**Comment by aahung** [comment]
Fixed in 1.21.0, please let us know if the issue still persists.
